### PR TITLE
RSDK-1937 Move trossen arm out of rdk

### PR DESCRIPTION
**Pull request by randhid**
This PR deletes the trossen arm and gripper packages. The arm will be modularized and debugged in further work: RSDK-2112. 

When removing the trossen package and kinematics, I discovered that many `referenceframe` and `motionplan` tests use the 5dof `vx300` and 6dof `wx-250` kinematics models for tests. I replaced the model in the 5dof test with a yahboom kinematics file (a 5dof arm), and models in the 6dof tests with an xarm6 kinematics file (a 6dof arm). Both files are located in their respective arm directories.

There were a few areas where a concatenated "test" trossen arm file was used with less than 5dof (information up to the elbow), and a trailing varm  (3dof) kinematics file was found (this model was removed from rdk months ago). I removed both and replaced them with other kinematics files from `referenceframe/testjson`, but can re-add them if necessary. I'm

### CODE REVIEW

File: motionplan/constraint_test.go [hunk 1]
```diff
@@ -20,7 +20,7 @@ import (
 func TestIKTolerances(t *testing.T) {
 	logger := golog.NewTestLogger(t)
 
-	m, err := frame.ParseModelJSONFile(utils.ResolveFile("referenceframe/testjson/varm.json"), "")
+	m, err := frame.ParseModelJSONFile(utils.ResolveFile("referenceframe/testjson/ur5eDH.json"), "")
```
Comment (randhid) [inline motionplan/constraint_test.go:23]: @raybjork, here's one place where I took out the varm as well. Planning changed from a 2dof -> 6dof system.

File: referenceframe/model_json_test.go [hunk 2]
```diff
@@ -14,10 +14,8 @@ import (
 // So we'll just check that we read in the right number of joints.
 func TestParseJSONFile(t *testing.T) {
 	goodFiles := []string{
-		"components/arm/trossen/trossen_wx250s_kinematics.json",
-		"components/arm/trossen/trossen_wx250s_test.json",
+		"components/arm/xarm/xarm6_kinematics.json",
```
[the remark this answers is not in the document]
Comment (raybjork) [reply referenceframe/model_json_test.go:17]: Good point.  Actually, could you just add the dofbot and xarm7 model jsons in this PR? They should just pass the test out of the box, and if they don't you can ignore this comment

**Review by stevebriskin — APPROVED** [review]
lgtm assuming a viam-labs pr is coming.

**Review by raybjork — APPROVED** [review]

**Comment by Otterverse** [comment]
Just realized one thing got overlooked... should've run `go mod tidy` to remove the dynamixel package dependency too. Oh well, it'll get tidied by another PR before too long.

**Comment by randhid** [comment]
Pulled the trigger a bit fast. I'll remember to run on whatever m  next pr is.

**Pull request merged**